File: 0704/704.binary-search.java
/*
 * @lc app=leetcode id=704 lang=java
 *
 * [704] Binary Search
 */

// @lc code=start
class Solution {
    public int search(int[] nums, int target) {

        int result = -1;

        for (int k = 0; k < nums.length; k++){
            if (nums[k]==target){
                result = k;
                break;
            }
            if (nums[k] > target){
                return -1;
            }
        }
        
        return result;
    }
}
// @lc code=end

File: 0704/704.binary-search.py
#
# @lc app=leetcode id=704 lang=python
#
# [704] Binary Search
#

# @lc code=start
class Solution(object):
    def search(self, nums, target):

        result = -1

        for idx, x in enumerate(nums):
            if x == target:
                result = idx
                break
            if x > target:
                return -1

        return result
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """

# @lc code=end
